src/Runtime/Runtime/Core/Rendering/INTERNAL_PopupsManager.cs

```diff
@@ -260,16 +260,16 @@ public static Point GetUIElementAbsolutePosition(UIElement element)
             return new Point();
         }
 
-        public static void EnsurePopupStaysWithinScreenBounds(Popup popup)
+        public static void EnsurePopupStaysWithinScreenBounds(Popup popup, Size size = new Size())
```

Comment:
This is going to force the popup size to be (0, 0) whenever size uses the default value (`new Size()`). You can use this instead (and make the appropriates changes lines 271 and 272)
```suggestion
        public static void EnsurePopupStaysWithinScreenBounds(Popup popup, double forcedWidth = double.NaN, double forcedHeight = double.NaN)
```